docs/openshift_horizontal_scaling.md

```diff
@@ -0,0 +1,105 @@
+guide](https://github.com/3scale/3scale-operator/blob/master/doc/template-user-guide.md).
+
+## Openshift HPA
+
+Openshift Horizontal Pod Autoscaling (HPA) allows to scale pods horizontally by
+specifying the minimum and maximum number of pods that should be deployed and
+the target CPU utilization. For more details check the [official Openshift
+docs](https://docs.openshift.com/container-platform/4.1/nodes/pods/nodes-pods-autoscaling.html).
+
+## Async
+
+Apisonator can be configured to use a non-blocking Redis client. With this
+feature enabled, a single process should use a whole CPU core when properly
+tuned. This means that we can enable Openshift HPA using the CPU utilization as
+the scaling metric.
+
+To learn more about how the async Redis client works, check this [design
+doc](./async.md).
+
+## Setup
+
+### Redis databases
+
+The standard 3scale deployment uses a single Redis pod and sets both the resque
+DB and the data DB in the same Redis process under different DB indexes. This is
+not supported by the async redis library, so we need to either place both DBs in
+the same Redis process or use two separate ones. To configure this, change the
+`CONFIG_REDIS_PROXY` and `REDIS_QUEUES_URL` envs in the listener and worker
+deployment configs.
+
+### Enable async
+
+- Set the `CONFIG_REDIS_ASYNC` to true in the listener and worker deployment configs.
+- Configure listeners to use the Falcon web-server instead of Puma. In order to
+do this, modify the command in the listener deployment config. Replace this:
+```yaml
+spec:
+  containers:
+    - args:
+        - bin/3scale_backend
+        - start
+        - '-e'
+        - production
+        - '-p'
+        - '3000'
+        - '-x'
+        - /dev/stdout
+```
+
+with:
+```yaml
+spec:
+  containers:
+    - args:
+        - bin/3scale_backend
+        - '-s'
+        - falcon
+        - start
+        - '-e'
+        - production
+        - '-p'
+        - '3000'
+        - '-x'
+        - /dev/stdout
+```
+- Set the `PUMA_WORKERS` env to 1 in the listener deployment. This will deploy 1
+Falcon worker per pod.
+
+### Set up horizontal pod autoscaling
+
+We need to decide the target CPU usage. It is defined as a percentage of the
+requested CPU cores. In a default installation, the backend-listener pod
+requests 0.5 CPU cores. In an ideal case, we know that both the listener and the
+worker should use 1 CPU core when using the async redis lib. However, we need to
+leave some margin for possible inefficiencies, so we can target 0.8 CPU cores,
+for example. 0.8/0.5 = 1.6, so we need to define a CPU % of 160 for the
+listener. In a default installation, the backend-worker pod requests 0.150 CPU
+cores. Following the same logic, we need to define a target CPU of 533
+(0.8/0.150 = 5.33). Pay attention to the CPU request of each pod, as they might
+change between versions, and adapt the target CPU % values accordingly.
```

Comment:
So here comes the logical question: why would we want to scale so much more the workers CPU demand (to 0.8) when we only request initially 0.150 millicores (when compared to the listeners which start way higher)? We could add something, even if a little vague, about why this is the case.

Reply:
Even if you request only 0.150, you can ask for a limit of 1.

Reply:
In my opinion, it's a bit confusing to define the target CPU as a function of the pod request. It'd be more intuitive to just define the number of cores.

Reply:
And the docs are not helping. Will look them up in k8s docs and see if there is any clearer explanation.